Fix running e2e with 'Completed' kube-system pods

ivan4th opened the pull request:
As of now, e2e runner keeps waiting for pods in `kube-system` namespace to be "Running and Ready" if there are any pods in `Completed` state in that namespace.
This for example happens after following [Kubernetes Hosted Installation](http://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/#kubernetes-hosted-installation) instructions for Calico, making it impossible to run conformance tests against the cluster. It's also to possible to reproduce the problem like that:
```
$ cat testjob.yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: tst
  namespace: kube-system
spec:
  template:
    metadata:
      name: tst
    spec:
      containers:
      - name: tst
        image: busybox
        command: ["echo",  "test"]
      restartPolicy: Never
$ kubectl create -f testjob.yaml
$ go run hack/e2e.go -v --test --test_args='--ginkgo.focus=existing\s+RC'
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37325)
<!-- Reviewable:end -->

ixdy commented on test/e2e/framework/util.go:
```diff
@@ -547,7 +547,7 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods int32, ti
 				continue
 			}
-			if res, err := testutils.PodRunningReady(&pod); res && err == nil {
+			if res, err := testutils.PodRunningReadyOrSucceeded(&pod); res && err == nil {
```
if we do this, should probably update the name of the function, too.

ivan4th commented:
@ixdy I took closer look at the code and it looks like that the original behavior may be desired in mesos case. Also, what should really be done perhaps is just not counting Succeeded pods, because minPods means the number of Running and Ready pods, and Succeeded pods should not be counted as such. So, I added `skipSucceeded` arg to the function which is always set to true except for the mesos case. I also added a check so it doesn't go into long (e.g. 10 minutes) loop in case if no succeeded pods are expected but some of the pods are in that state, returning an error immediately instead.

ixdy commented:
```
W1122 17:58:06.559] test/e2e/framework/util.go:558: missing argument for Logf("%s"): format reads arg 1, have only 0 args
```

ivan4th commented:
Thanks, fixed.

Commit:
Fix running e2e with 'Completed' kube-system pods

ixdy commented:
/lgtm

ixdy commented:
@k8s-bot test this

ivan4th commented:
The problem looks like #27680

@k8s-bot cvm gce e2e test this
